algorithms/CPlusPlus/sliding-window.cpp

```diff
@@ -0,0 +1,56 @@
+/*
+Description: A program to find target sub string in given string 
+
+Approach: Using sliding window technique to compare every possible substring with the target string.
+Also Implementing the variable length sliding window algorithm where user can give size of window and target string of same size
+
+Time complexity: O(n)
+*/
+
+
+#include <iostream>
+#include <string>
+
+using namespace std;
+
+void sliding(string s,string target){
+    int window_size=5;
+    
+    for(int i=0;i<s.size();i++){
+        
+        string value = s.substr(i,window_size);
+       
+        if(target==value){
+            cout<<target<<" found at "<<i<<" and "<<i+window_size<<endl;
+        }
+    }
+}
+
+void sliding_with_variable_length(string s,string target,int wins){
```

Comment:
giving user the ability to alter sliding windows size opens the opportunity for them to input different sizes which is is not necessarily the size of target string. How will the program handle such case ?

Reply:
so for  that can I initialize window size with length of the target string instead of explicitly tacking user input for window size?

Reply:
@Amit-exe indeed, that would also remove the need of having 2 overloads for the same function.